src/ruby/lib/grpc/generic/rpc_server.rb

```diff
@@ -407,7 +413,13 @@ def loop_handle_server_calls
       request_call_tag = Object.new
       until stopped?
         deadline = from_relative_time(@poll_period)
-        an_rpc = @server.request_call(@cq, request_call_tag, deadline)
+        begin
+          an_rpc = @server.request_call(@cq, request_call_tag, deadline)
+        rescue Core::CallError, RuntimeError => e
+          # can happen during server shutdown
```

Comment:
odd - this block is there because that's what I've observed.  Are you sure that success=true and call=null occur for shutdown even when the server is shutdown from a separate thread ?